simd/src/Kokkos_Simd_AVX512.hpp

```diff
@@ -0,0 +1,928 @@
+      : m_value(static_cast<__m512i>(other)) {}
+  KOKKOS_HOST_FORCEINLINE_FUNCTION reference operator[](std::size_t i) {
+    return reinterpret_cast<value_type*>(&m_value)[i];
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION value_type operator[](std::size_t i) const {
+    return reinterpret_cast<value_type const*>(&m_value)[i];
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd operator*(simd const& other) const {
+    return _mm512_mullo_epi64(m_value, other.m_value);
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd operator+(simd const& other) const {
+    return _mm512_add_epi64(m_value, other.m_value);
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd operator-(simd const& other) const {
+    return _mm512_sub_epi64(m_value, other.m_value);
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd operator>>(unsigned int rhs) const {
+    return _mm512_srli_epi64(m_value, rhs);
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd operator>>(
+      simd<std::int32_t, simd_abi::avx512_fixed_size<8>> const& rhs) const {
+    return _mm512_srlv_epi64(m_value,
+                             _mm512_cvtepi32_epi64(static_cast<__m256i>(rhs)));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd operator<<(unsigned int rhs) const {
+    return _mm512_slli_epi64(m_value, rhs);
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd operator<<(
+      simd<std::int32_t, simd_abi::avx512_fixed_size<8>> const& rhs) const {
+    return _mm512_sllv_epi64(m_value,
+                             _mm512_cvtepi32_epi64(static_cast<__m256i>(rhs)));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd operator&(simd const& other) const {
+    return _mm512_and_epi64(m_value, other.m_value);
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd operator|(simd const& other) const {
+    return _mm512_or_epi64(m_value, other.m_value);
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION constexpr explicit operator __m512i() const {
+    return m_value;
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION mask_type
+  operator<(simd const& other) const {
+    return mask_type(_mm512_cmplt_epu64_mask(m_value, other.m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION mask_type
+  operator>(simd const& other) const {
+    return mask_type(_mm512_cmplt_epu64_mask(other.m_value, m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION mask_type
+  operator<=(simd const& other) const {
+    return mask_type(_mm512_cmple_epu64_mask(m_value, other.m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION mask_type
+  operator>=(simd const& other) const {
+    return mask_type(_mm512_cmple_epu64_mask(other.m_value, m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION mask_type
+  operator==(simd const& other) const {
+    return mask_type(_mm512_cmpeq_epu64_mask(m_value, other.m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION mask_type
+  operator!=(simd const& other) const {
+    return mask_type(_mm512_cmpneq_epu64_mask(m_value, other.m_value));
+  }
+};
+
+KOKKOS_HOST_FORCEINLINE_FUNCTION
+simd<std::uint64_t, simd_abi::avx512_fixed_size<8>> condition(
```

Comment:
Is this standardized? Otherwise,
```suggestion
simd<std::uint64_t, simd_abi::avx512_fixed_size<8>> blend(
```
and some comments what this does would be good. :slightly_smiling_face:

Reply:
This is not standardized. The latest conversation related to standardization:
https://github.com/mattkretz/std-simd-feedback/issues/17
Says that they would rely on being able to overload `operator?:`.

This function is used a lot in my physics application and in others at Sandia, so a wider conversation would be needed to rename it. I chose `condition` because `operator?:` is sometimes called the conditional operator. Similarly, `std::conditional` chooses one of two types based on a condition.

Reply:
If this is already used a lot, that's fine with me.
I just found it quite difficult to understand what this is doing without looking up the intrinsic you are using and without any comments.

Reply:
Good point, I can add a comment on overloads and make sure to prominently document it in the future PR to the Kokkos Wiki